contracts/src/v0.8/ccip/pools/GHO/UpgradeableTokenPool.sol

```diff
@@ -0,0 +1,611 @@
+  function _setRemotePool(uint64 remoteChainSelector, bytes memory remotePoolAddress) internal {
+    if (remotePoolAddress.length == 0) {
+      revert ZeroAddressNotAllowed();
+    }
+
+    bytes32 poolHash = keccak256(remotePoolAddress);
+
+    // Check if the pool already exists.
+    if (!s_remoteChainConfigs[remoteChainSelector].remotePools.add(poolHash)) {
+      revert PoolAlreadyAdded(remoteChainSelector, remotePoolAddress);
+    }
+
+    // Add the pool to the mapping to be able to un-hash it later.
+    s_remotePoolAddresses[poolHash] = remotePoolAddress;
+
+    emit RemotePoolAdded(remoteChainSelector, remotePoolAddress);
+  }
+
+  // ================================================================
+  // │                        Rate limiting                         │
+  // ================================================================
+
+  /// @dev The inbound rate limits should be slightly higher than the outbound rate limits. This is because many chains
+  /// finalize blocks in batches. CCIP also commits messages in batches: the commit plugin bundles multiple messages in
+  /// a single merkle root.
+  /// Imagine the following scenario.
+  /// - Chain A has an inbound and outbound rate limit of 100 tokens capacity and 1 token per second refill rate.
+  /// - Chain B has an inbound and outbound rate limit of 100 tokens capacity and 1 token per second refill rate.
+  ///
+  /// At time 0:
+  /// - Chain A sends 100 tokens to Chain B.
+  /// At time 5:
+  /// - Chain A sends 5 tokens to Chain B.
+  /// At time 6:
+  /// The epoch that contains blocks [0-5] is finalized.
+  /// Both transactions will be included in the same merkle root and become executable at the same time. This means
+  /// the token pool on chain B requires a capacity of 105 to successfully execute both messages at the same time.
+  /// The exact additional capacity required depends on the refill rate and the size of the source chain epochs and the
+  /// CCIP round time. For simplicity, a 5-10% buffer should be sufficient in most cases.
+
+  /// @notice Sets the rate limiter admin address.
+  /// @dev Only callable by the owner.
+  /// @param rateLimitAdmin The new rate limiter admin address.
+  function setRateLimitAdmin(address rateLimitAdmin) external onlyOwner {
+    s_rateLimitAdmin = rateLimitAdmin;
+    emit RateLimitAdminSet(rateLimitAdmin);
+  }
+
+  /// @notice Gets the rate limiter admin address.
+  function getRateLimitAdmin() external view returns (address) {
+    return s_rateLimitAdmin;
+  }
+
+  /// @notice Consumes outbound rate limiting capacity in this pool
+  function _consumeOutboundRateLimit(uint64 remoteChainSelector, uint256 amount) internal {
+    s_remoteChainConfigs[remoteChainSelector].outboundRateLimiterConfig._consume(amount, address(i_token));
+  }
+
+  /// @notice Consumes inbound rate limiting capacity in this pool
+  function _consumeInboundRateLimit(uint64 remoteChainSelector, uint256 amount) internal {
```

Comment:
Small nit: We could go by solidity style guidelines and put these internal functions at the end of the section, and shift external view to appear after the external non-view https://docs.soliditylang.org/en/v0.8.21/style-guide.html#order-of-functions

Reply:
i believe we should be following the base CLL order to keep diff's minimal for review and audit. lmk if you feel otherwise, not a strong opinion here

Reply:
That's cool, but on default branch, we have these internals showing up first in this section, so should we stick to that?: https://github.com/aave/ccip/blob/ccip-gho/contracts/src/v0.8/ccip/pools/GHO/UpgradeableTokenPool.sol#L204

Reply:
that's not the base branch (it's got CCIP 1.4), base is `ccip-develop`, pls see PR description for reasoning